src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java

```diff
@@ -38,81 +41,76 @@
     @Autowired
     private UserProfileClient userProfileClient;
 
-    @Autowired
-    private SyncJobRepository syncJobRepository;
-
     @Value("${loggingComponentName}")
-    protected String loggingComponentName;
+    private String loggingComponentName;
 
-    public void updateUserProfile(String searchQuery, String bearerToken, String s2sToken, List<IdamClient.User> users) throws UserProfileSyncException {
-        log.info(loggingComponentName, "Inside updateUserProfile:: ");
+    public ProfileSyncAudit updateUserProfile(String searchQuery, String bearerToken, String s2sToken, Set<IdamClient.User> users, ProfileSyncAudit syncAudit) throws UserProfileSyncException {
+        log.info("{}:: Inside updateUserProfile::{} ", loggingComponentName);
+        List<ProfileSyncAuditDetails> profileSyncAuditDetails = new ArrayList<>();
         users.forEach(user -> {
             Optional<GetUserProfileResponse> userProfile = userAcquisitionService.findUser(bearerToken, s2sToken, user.getId());
 
             if (userProfile.isPresent()) {
-                Map<String, Boolean> status = new HashMap<String, Boolean>();
-                status.put(IdamStatus.ACTIVE.name(), user.isActive());
-                status.put(IdamStatus.PENDING.name(), user.isPending());
+                StringBuilder sb = new StringBuilder();
+                sb.append(user.isActive());
+                sb.append(user.isPending());
                 UserProfile updatedUserProfile = UserProfile.builder()
                         .email(user.getEmail())
                         .firstName(user.getForename())
                         .lastName(user.getSurname())
-                        .idamStatus(idamStatusResolver().get(status) != null ? idamStatusResolver().get(status).name() : IdamStatus.SUSPENDED.name())
+                        .idamStatus(resolveIdamStatus(sb))
                         .build();
 
                 try {
-
-                    syncUser(bearerToken, s2sToken, user.getId(), updatedUserProfile);
+                    profileSyncAuditDetails.add(syncUser(bearerToken, s2sToken, user.getId(), updatedUserProfile,syncAudit));
```

Comment:
this is not the best practice. SyncUser is the main method which syncs the user status with User profile. However it is called an argument to the audit method which makes it hard to understand the input being passed and also kind of give impression that profileSyncAuditDetails.add() is doing main business logic
It makes it bit more difficult to understand and maintain

Reply:
i have added the comment like what theline is doing to avode confusion. I have tried but no other way